app/renderer/ui/components/main/sections/wallet/tabs/receive/index.tsx

```diff
@@ -1,28 +1,177 @@
+} from "app/common/runtimeTypes/ipc/address"
+import { assertNever } from "app/common/utils"
+import { KEYCHAIN_INDICES } from "app/renderer/constants/wallet"
 
 const styles = require("./style.scss")
 
+/**
+ * Props that contain configuration
+ *
+ * @interface Props
+ */
+interface Props {
+  selectedAccount: number,
+  account: Account,
+  services: Services,
+  saveFinalKey: Function
+}
+
 /**
  * TabReceive component
  *
  * @class TabReceive
  * @extends {TabComponent<any>}
  */
-class TabReceive extends TabComponent<any> {
+class TabReceive extends TabComponent<any & Props> {
+
+  /**
+   * Local state to form inputs and loading state
+   */
+  public state = {
+    label: "",
+    amount: "",
+    expires: "",
+    loading: false,
+    errorMessage: ""
+  }
+
+  /**
+   * Method to perform an async call to renderer API to generate a Final Key and
+   * resolve/reject a promise depending on the result of that call
+   * @param params
+   */
+  private generateAddress = async (label: string, amount: string, expires: string) => {
+    return new Promise<FinalKey>((resolve, reject) => {
+      this.parseGenerateAddressParams(label, amount, expires)
+        .then(this.apiGenerateAddress)
+        .then((response: GenerateAddressResponse) => {
+          switch (response.kind) {
+            case "SUCCESS":
+              resolve(response.finalKey)
+              break
+            case "ERROR":
+              reject(generateAddressErrorMessages[response.error] || "Unknown error")
+              break
+            default:
+              assertNever(response)
+          }
+        })
+        .catch((error) => {
+          reject(`GENERATE_ADDRESS_UNEXPECTED_ERROR ${error.message}`)
+        })
+    })
+  }
+
+  /**
+   * Parse generate address params and returns an object with valid params
+   */
+  private parseGenerateAddressParams = async (
+    pLabel: string,
+    pAmount: string,
+    pExpires: string
+  ): Promise<GenerateAddressParams> => {
+    let result: GenerateAddressParams = { account: this.props.selectedAccount }
+    const label = pLabel.trim()
+    if (label.length > 0) {
+      result = { ...result, label }
+    }
+    const requestedAmount = Number.parseFloat(pAmount)
+    if (!Number.isNaN(requestedAmount)) {
+      result = { ...result, requestedAmount }
+    }
+    const expirationDate = new Date(pExpires).getTime() / 1000
+    if (!Number.isNaN(expirationDate)) {
+      result = { ...result, expirationDate: Math.floor(expirationDate) }
+    }
+
+    return result
+  }
+
+  /**
+   * Calls API for generating external address
+   */
+  private apiGenerateAddress = async (params: {
+    account: number,
+    label?: string,
+    requestedAmount?: number,
+    expirationDate?: number
+  }): Promise<GenerateAddressResponse> => {
+    const apiParams: GenerateAddressParams = {
+      ...params,
+      account: params.account
+    }
+
+    return api.generateAddress(this.props.services.apiClient, apiParams)
```

Comment:
I think that the type of the `params` parameter may be just `GenerateAddressParams` and so if typed like that, `params` could be passed directly to `api.generateAddress(this.props.services.apiClient, params)`